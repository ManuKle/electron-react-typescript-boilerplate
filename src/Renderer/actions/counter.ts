import { RootState } from "@red/index";
 
export type counterActions = increment | decrement;
 
interface increment {
  type: string;
}
 
interface decrement {
  type: string;
}
 
const increment = () => ({
  type: actions.INCREMENT_COUNTER
});
 
const decrement = () => ({
  type: actions.DECREMENT_COUNTER
});
 
const incrementIfOdd = () => {
  return (
    dispatch: (action: counterActions) => void,
    getState: () => RootState
  ) => {
    const { counter } = getState().counter;
 
    if (counter % 2 === 0) {
      return;
    }
 
    dispatch(increment());
  };
};

const incrementAsync = (delay: number = 1000) => {
  return (dispatch: (action: counterActions) => void) => {
    setTimeout(() => {
      dispatch(increment());
    }, delay);
  };
};
 
export const actions = {
  INCREMENT_COUNTER: "INCREMENT_COUNTER",
  DECREMENT_COUNTER: "DECREMENT_COUNTER"
}
export const actionCreators = {
  increment,
  decrement,
  incrementIfOdd,
  incrementAsync
};
