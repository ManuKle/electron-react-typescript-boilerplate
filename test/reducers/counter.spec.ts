import { counterReducer } from '../../src/Renderer/reducers/counter';
import { actions } from '../../src/Renderer/actions/counter';

describe('reducers', () => {
  describe('counterReducer', () => {
    it('should handle initial state', () => {
      expect(counterReducer(undefined, {})).toMatchSnapshot();
    });

    it('should handle INCREMENT_COUNTER', () => {
      expect(counterReducer(1, { type: actions.INCREMENT_COUNTER })).toMatchSnapshot();
    });

    it('should handle DECREMENT_COUNTER', () => {
      expect(counterReducer(1, { type: actions.DECREMENT_COUNTER })).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(counterReducer(1, { type: 'unknown' })).toMatchSnapshot();
    });
  });
});
