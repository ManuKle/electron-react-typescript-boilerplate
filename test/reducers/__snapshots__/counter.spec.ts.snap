// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers counterReducer should handle DECREMENT_COUNTER 1`] = `
Object {
  "counter": NaN,
}
`;

exports[`reducers counterReducer should handle INCREMENT_COUNTER 1`] = `
Object {
  "counter": NaN,
}
`;

exports[`reducers counterReducer should handle initial state 1`] = `
Object {
  "counter": 0,
}
`;

exports[`reducers counterReducer should handle unknown action type 1`] = `1`;
